### PR TITLE
Require the same FileSystem provider

### PR DESCRIPTION
**Pull request by hakonhall**
TL;DR  Require all paths with NodeAgentContext[Impl] to use the same file system provider, and add a `FileSystem fileSystem()` method.

NodeAgentContext[Impl] requires paths to have specific providers, otherwise a ProviderMismatchException will be thrown, which it often does when adding tests involving ths class.  The current API is best described by defining a few terms:

* Let PathDefault be a Path with a file system provider matching that of FileSystems.getDefault().provider().
* Let PathContainerStorage be a Path with the same file system provider as the context's pathToContainerStorage constructor param
* Let PathVespaHome be a Path with the same file system provider as the context's pathToVespaHome constructor param.

The context methods handling paths have these requirements:

* `Path pathOnHostFromPathInNode(Path pathInNode)` maps a PathDefault to PathContainerStorage
* `Path pathInNodeFromPathOnHost(Path pathOnHost)` maps a PathContainerStorage to PathDefault
* `Path pathInNodeUnderVespaHome(Path relativePath)` maps a PathVespaHome to PathVespaHome
* `Path rewritePathInNodeForWantedDockerImage(Path path)` maps a PathDefault to PathDefault, and requires PathVespaHome == PathDefault, if it is to work under all circumstances.

This PR requires all paths with NAC[I] to have the same provider, otherwise we'll detect this and throw as early as possible.

### CODE REVIEW

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java [hunk 1]
```diff
@@ -296,8 +300,9 @@ public String getCloudNativeRegionName() {
                             return getId().region().value();
                         }
                     }),
-                    Optional.ofNullable(pathToContainerStorage).orElseGet(() -> fileSystem.getPath("/home/docker")),
-                    Optional.ofNullable(pathToVespaHome).orElseGet(() -> fileSystem.getPath("/opt/vespa")),
+                    fileSystem,
+                    fileSystem.getPath("/home/docker/container-storage"),
```
Comment (freva) [inline node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java:304]: The `container-storage` addition means more typing in tests... :(

**Review by freva — APPROVED** [review]

File: node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java (hunk 1 above)
Comment (hakonhall) [reply node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/nodeagent/NodeAgentContextImpl.java:304]: yes, but it matches the expected path.  I was hoping this could lead to simplifying the NodeAgentContextImpl constructor by removing 2 path parameters - but I ended up keeping them.
